src/uu/tee/src/tee.rs

```diff
@@ -181,38 +176,41 @@ fn tee(options: &Options) -> Result<()> {
         _ => Ok(()),
     };
 
-    if res.is_err() || output.flush().is_err() || output.error_occurred() {
+    if had_open_errors || res.is_err() || output.flush().is_err() || output.error_occurred() {
         Err(Error::from(ErrorKind::Other))
     } else {
         Ok(())
     }
 }
 
+/// Tries to open the indicated file and return it. Reports an error if that's not possible.
+/// If that error should lead to program termination, this function returns Some(Err()),
+/// otherwise it returns None.
 fn open(
-    name: String,
+    name: &str,
     append: bool,
     output_error: Option<&OutputErrorMode>,
-) -> Result<Box<dyn Write>> {
+) -> Option<Result<NamedWriter>> {
     let path = PathBuf::from(name.clone());
```

Comment:
please see
ERROR: `cargo clippy`: call to `.clone()` on a reference in this situation does nothing (file:'src/uu/tee/src/tee.rs', line:194)

```suggestion
    let path = PathBuf::from(name);
```
(i didn't test)

Reply:
You're right!